templates/privacyidea.xhtml

```diff
@@ -0,0 +1,98 @@
+		<f:viewParam name="login_hint" value="#{loginAction.loginHint}" />
+	</f:metadata>
+	<ui:define name="pageTitle">
+		<h:outputText value="privacyIDEA - oxAuth" />
+	</ui:define>
+	<ui:define name="body">
+		<div class="container">
+			<h:panelGroup rendered="true">
+				<div class="login_bx_1"
+					style="border-radius: 10px; margin-top: 0px; background: white; border: 1px solid #008b8b;">
+					<div class="row">
+						<h:messages class="text-center"
+							style="color:#4ca4ec;margin:5px;margin-left:20px; font-size:1vw;" infoClass="text-center"
+							errorClass="text-center" />
+					</div>
+					<h:form id="loginForm" style="padding:30px;" prependId="false">
+						<div class="form-group row"></div>
+						<div class="col-sm-offset-2 offset-md-2 col-sm-8 col-md-8">
+							<p>
+								<h:outputText value="#{identity.getWorkingParameter('transaction_message')}" />
+							</p>
+						</div>
+						<div class="form-group row">
+							<div class="col-sm-offset-2 offset-md-2 col-sm-8 col-md-8">
+								<h:inputSecret placeholder="OTP" colMd="10" id="otp" name="otp" labelColMd="2"
+									value="#{otp}" autocomplete="off" styleClass="form-control" tyle="width:100%">
+								</h:inputSecret>
+							</div>
+						</div>
+						<div class="form-group row"></div>
+						<div class="form-group row">
+							<div class="col-sm-offset-2 offset-md-2 col-sm-8 col-md-8">
+								<h:commandButton id="loginButton" style="background-color: #00BE79; color:white;"
+									styleClass="btn col-sm-12" value="    #{msgs['login.login']}"
+									iconAwesome="fa-sign-in" action="#{authenticator.authenticate}" />
+							</div>
+						</div>
+						<div class="form-group row"></div>
+						<div class="form-group row">
+							<div class="col-sm-offset-2 offset-md-2 col-sm-8 col-md-8">
+								<h:commandButton id="changeModeButton" style="background-color: #00BE79; color:white;"
+									styleClass="btn col-sm-12" value="    Use PushToken" onclick="changeMode()"
+									iconAwesome="fa-sign-in" action="#{authenticator.authenticate}" />
+							</div>
+						</div>
+						<h:inputHidden id="modeField" value="otp" />
+					</h:form>
+				</div>
+			</h:panelGroup>
+		</div>
+		<script>
+			function changeMode() {
+				var modeField = document.getElementById("modeField");
+				modeField.value = (modeField.value == "otp") ? "push" : "otp";
+			}
+
+			window.onload = function () {
+				var pushAvailable = "#{identity.getWorkingParameter('push_available')}";
+				var mode = "#{identity.getWorkingParameter('mode')}";
+				var otpAvailable = "#{identity.getWorkingParameter('otp_available')}";
+				var modeField = document.getElementById("modeField");
+				var changeModeButton = document.getElementById("changeModeButton");
+				var otpField = document.getElementById("otp");
+
+				if (mode == "otp") {
+					otpField.value = "";
+					otpField.focus();
+				}
+
+				if (!(mode == "push" || otpAvailable == "1")) {
```

Comment:
...except here. Is ``otpAvailable`` actually a string or an integer?

Reply:
it is a string.

Reply:
see above. *Could* ``otpAvailable`` be an integer?